contracts/TablelandTables.sol

```diff
@@ -99,11 +101,42 @@ contract TablelandTables is
     function _getPolicy(
         address caller,
         uint256 tableId
-    ) private returns (ITablelandController.Policy memory) {
+    ) private returns (TablelandPolicy memory) {
         address controller = _controllers[tableId];
         if (_isContract(controller)) {
-            return
+            // Try {ITablelandController.getPolicy(address, uint256)}.
+            try
                 ITablelandController(controller).getPolicy{value: msg.value}(
+                    caller,
+                    tableId
+                )
+            returns (TablelandPolicy memory policy) {
+                return policy;
+            } catch Error(string memory reason) {
+                // Controller reverted/required with a reason string. Bubble up the error.
+                revert(reason);
+            } catch (bytes memory err) {
+                // We are here for one of two reasons:
+                // 1. We hit an old controller that does not implement {ITablelandController.getPolicy(address, uint256)}.
+                // 2. The controller reverted w/o a reason string, e.g, a custom error, revert(), or require(condition).
+                // We can't differentiate between reverting/requiring w/o a reason string and a v0 controller.
+                // When a controller reverts/requires w/o a reason string it will be treated as a v0 controller,
+                // i.e., we will try to call {ITablelandControllerV0.getPolicy(address)}.
+
+                // Controller reverted with a custom error. Bubble it up.
+                if (err.length > 0) {
+                    assembly {
+                        revert(add(32, err), mload(err))
```

Comment:
see [here](https://ethereum.stackexchange.com/a/111187/104267) for an explanation. also [here](https://github.com/ethereum/solidity/issues/11278#issuecomment-1239574905).

Reply:
Bubbling up Custom Errors was a big piece of what I was missing before, so this looks great.